packages/@lwc/template-compiler/src/__tests__/fixtures/attributes/valid-hyphen-attribute/expected.js

```diff
@@ -0,0 +1,73 @@
+import _xButton from "x/button";
+import { registerTemplate } from "lwc";
+const stc0 = {
+  props: {
+    under_Hyphen: "bar",
+  },
+  key: 0,
+};
+const stc1 = {
+  props: {
+    PascalCase: "bar",
+  },
+  key: 1,
+};
+const stc2 = {
+  props: {
+    Hyphen: "bar",
+  },
+  key: 2,
+};
+const stc3 = {
+  props: {
+    hyphen3pecial: "bar",
+  },
+  key: 3,
+};
+const stc4 = {
+  props: {
+    hyphenSSpecial: "bar",
```

Comment:
This conversion looks surprising to me. After digging further, it appears to [be correct](https://german.stackexchange.com/questions/8958/frequency-of-the-majuscule-%c3%9f-%e1%ba%9e-in-modern-german/8960#8960). To me this opens the question of automatically to uppercasing non-latin characters. 

IMO, we should prevent this, the compiler should error out if an attribute name contains an `-` followed by a non-latin character. Same observation with numbers, what is the upper case character of a number character?

Reply:
> To me this opens the question of automatically to uppercasing non-latin characters

That ship might have sailed. There is no restriction today to restrict what character follows a hyphen. So we might break customers if we add new restriction. We will need to find usage data to answer this question affirmatively.

> IMO, we should prevent this, the compiler should error out if an attribute name contains an - followed by a non-latin character. 

Same as above, need data before adding this restriction.
Greek characters also can be upper cased. https://jsbin.com/gamoxep/edit?js,console

>Same observation with numbers, what is the upper case character of a number character?

`toUpperCase()` for numbers is a no-op. I have added a test case for that(L23 in this AST). 

tl;dr
There is a lack of standards here wrt to _attribute_ to _property_ reflection. LWC seems to automatically call toUpperCase() on any character following a hyphen. Some characters like(alpha, latin characters) have an upper case version and they work. For the rest of the characters like(numeric, emoji), toUpperCase() is a no-op.
The toUpperCase() method follows the Unicode Default Case Conversion algorithm. [Here's the full mapping table.](https://www.ibm.com/docs/en/i/7.2?topic=tables-unicode-lowercase-uppercase-conversion-mapping-table)

Here's what I found about other framework's stance about this
1. Lit: They [do not support](https://github.com/lit/lit-element/issues/29) this automatic kebab-case to camelCase
2. Vue.js supports both kebab-case and camelCase in the markup. They automatically convert kebab-case to camelCase properties. However, vue does not support hyphen char followed by a special char. For example: `props: ['hyphenÀpecial']` in a vue component be set using `<Comp hyphenÀpecial="hello"/>` and not `hyphen-àpecial`
3. Svelte markup supports camelCase attribute names. It does not automatically convert kebab-case to camelCase. For svelte components used as web components, camelCase attributes do not work. [1](https://community.webcomponents.dev/tnzk/svelte-for-web-components-development-pitfalls-and-workarounds-1405), [2](https://github.com/sveltejs/svelte/issues/875)

Reply:
@pmdartus I will proceed to merge this PR to enable some downstream testing. Hope we can continue this discussion and address it in followup PRs.

Reply:
You are correct @ravijayaramappa. Let's not make this more complex and stick to the Unicode case conversion.